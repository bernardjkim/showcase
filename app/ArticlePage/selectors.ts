import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { ArticleState } from './types';

/**
 * Direct selector to the articlePage state domain
 */

const selectArticlePageDomain = (state: any): ArticleState => state.get('articlePage', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.error,
  );

const makeSelectArticle = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.article,
  );

const makeSelectComments = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.comments,
  );

const makeSelectLikes = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.likes,
  );
/**
 * Default selector used by ArticlePage
 */

const makeSelectArticlePage = () =>
  createSelector(
    selectArticlePageDomain,
    substate => substate,
  );

export default makeSelectArticlePage;
export {
  selectArticlePageDomain,
  makeSelectArticle,
  makeSelectComments,
  makeSelectLikes,
  makeSelectLoading,
  makeSelectError,
};
