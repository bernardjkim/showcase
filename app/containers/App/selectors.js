import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectRouter = state => state.get('router');

const makeSelectLocation = () =>
  createSelector(selectRouter, routerState =>
    routerState.get('location').toJS(),
  );

/**
 * Direct selector to the global state domain
 */

const selectGlobalDomain = state => state.get('global', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('loading'));

const makeSelectError = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('error'));

const makeSelectUser = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('user'));

const makeSelectValidateToken = () =>
  createSelector(selectGlobalDomain, globalState =>
    globalState.get('validateToken'),
  );

/**
 * Default selector used by App
 */

const makeSelectApp = () =>
  createSelector(selectGlobalDomain, substate => substate.toJS());

export default makeSelectApp;
export {
  makeSelectLocation,
  selectGlobalDomain,
  makeSelectUser,
  makeSelectValidateToken,
  makeSelectLoading,
  makeSelectError,
};
