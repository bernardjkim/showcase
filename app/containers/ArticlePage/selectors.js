import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the articlePage state domain
 */

const selectArticlePageDomain = state => state.get('articlePage', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(selectArticlePageDomain, articleState =>
    articleState.get('loading'),
  );

const makeSelectError = () =>
  createSelector(selectArticlePageDomain, articleState =>
    articleState.get('error'),
  );

const makeSelectArticle = () =>
  createSelector(selectArticlePageDomain, articleState =>
    articleState.get('article'),
  );

/**
 * Default selector used by ArticlePage
 */

const makeSelectArticlePage = () =>
  createSelector(selectArticlePageDomain, substate => substate.toJS());

export default makeSelectArticlePage;
export {
  selectArticlePageDomain,
  makeSelectArticle,
  makeSelectLoading,
  makeSelectError,
};
