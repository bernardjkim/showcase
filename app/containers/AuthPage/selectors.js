import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the AuthPage state domain
 */

const selectAuthPageDomain = state => state.get('AuthPage', initialState);

/**
 * Other specific selectors
 */

/**
 * Default selector used by AuthPage
 */

const makeSelectAuthPage = () =>
  createSelector(selectAuthPageDomain, substate => substate.toJS());

export default makeSelectAuthPage;
export { selectAuthPageDomain };
