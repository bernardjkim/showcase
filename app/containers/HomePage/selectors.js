import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the homePage state domain
 */

const selectHomePageDomain = state => state.get('homePage', initialState);

/**
 * Other specific selectors
 */
const makeSelectArticle = () =>
  createSelector(selectHomePageDomain, homeState => homeState.get('article'));

/**
 * Default selector used by HomePage
 */

const makeSelectHomePage = () =>
  createSelector(selectHomePageDomain, substate => substate.toJS());

export default makeSelectHomePage;
export { selectHomePageDomain, makeSelectArticle };
