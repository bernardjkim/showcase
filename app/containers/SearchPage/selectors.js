import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the searchPage state domain
 */

const selectSearchPageDomain = state => state.get('searchPage', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(selectSearchPageDomain, searchState =>
    searchState.get('loading'),
  );

const makeSelectError = () =>
  createSelector(selectSearchPageDomain, searchState =>
    searchState.get('error'),
  );

const makeSelectArticles = () =>
  createSelector(selectSearchPageDomain, searchState =>
    searchState.get('articles'),
  );

/**
 * Default selector used by SearchPage
 */

const makeSelectSearchPage = () =>
  createSelector(selectSearchPageDomain, substate => substate.toJS());

export default makeSelectSearchPage;
export {
  selectSearchPageDomain,
  makeSelectArticles,
  makeSelectError,
  makeSelectLoading,
};
