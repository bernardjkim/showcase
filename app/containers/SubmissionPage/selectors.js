import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the submissionPage state domain
 */

const selectSubmissionPageDomain = state =>
  state.get('submissionPage', initialState);

/**
 * Other specific selectors
 */
const makeSelectLoading = () =>
  createSelector(selectSubmissionPageDomain, submissionState =>
    submissionState.get('loading'),
  );
const makeSelectError = () =>
  createSelector(selectSubmissionPageDomain, submissionState =>
    submissionState.get('error'),
  );

/**
 * Default selector used by SubmissionPage
 */

const makeSelectSubmissionPage = () =>
  createSelector(selectSubmissionPageDomain, substate => substate.toJS());

export default makeSelectSubmissionPage;
export { selectSubmissionPageDomain, makeSelectError, makeSelectLoading };
