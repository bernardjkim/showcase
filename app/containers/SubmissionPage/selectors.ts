import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { SubmissionState } from './types';

/**
 * Direct selector to the submissionPage state domain
 */

const selectSubmissionPageDomain = (state: any): SubmissionState => {
  return state.get('submissionPage', initialState);
}

/**
 * Other specific selectors
 */
const makeSelectLoading = () =>
  createSelector(
    selectSubmissionPageDomain,
    // submissionState => submissionState.get('loading'),
    submissionState => submissionState.loading,
  );
const makeSelectError = () =>
  createSelector(
    selectSubmissionPageDomain,
    // submissionState => submissionState.get('error'),
    submissionState => submissionState.error,
  );
const makeSelectSubmissionSuccess = () =>
  createSelector(
    selectSubmissionPageDomain,
    // submissionState => submissionState.get('submissionSuccess'),
    submissionState => submissionState.submissionSuccess,
  );

/**
 * Default selector used by SubmissionPage
 */

const makeSelectSubmissionPage = () =>
  createSelector(
    selectSubmissionPageDomain,
    // substate => substate.toJS(),
    substate => substate,
  );

export default makeSelectSubmissionPage;
export {
  selectSubmissionPageDomain,
  makeSelectError,
  makeSelectLoading,
  makeSelectSubmissionSuccess,
};

