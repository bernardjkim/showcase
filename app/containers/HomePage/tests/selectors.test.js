import { fromJS } from 'immutable';
import { selectHomePageDomain, makeSelectArticle } from '../selectors';

const initialState = fromJS({
  loading: false,
  error: false,
  article: false,
});

describe('selectHomePageDomain', () => {
  it('should select the home page state', () => {
    expect(selectHomePageDomain(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectArticle', () => {
  const articleSelector = makeSelectArticle();
  it('should select the home page state', () => {
    expect(articleSelector(initialState)).toMatchSnapshot();
  });
});
