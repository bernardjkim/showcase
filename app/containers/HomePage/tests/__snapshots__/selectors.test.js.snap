// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectArticle should select the article state 1`] = `false`;

exports[`makeSelectError should select the error state 1`] = `false`;

exports[`makeSelectLoading should select the loading state 1`] = `false`;

exports[`selectHomePageDomain should select the home page state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "article": false,
}
`;
